saml2idp/views.py

```diff
@@ -101,8 +109,18 @@ def logout(request):
     though it's technically not SAML 2.0).
     """
     auth.logout(request)
-    tv = {}
-    return render_to_response('saml2idp/logged_out.html', tv,
+
+    redirect_url = request.GET.get('redirect_to', '')
+    if redirect_url:
+
+        try:
+            validator = URL_VALIDATOR(redirect_url)
+        except ValidationError:
+            pass
+        else:
+            return HttpResponseRedirect(redirect_url)
+
+    return render_to_response('saml2idp/logged_out.html', {},
```

Comment:
Assuming that `URL_VALIDATOR()` will reject the empty string we could just do this which would remove one level of nesting and make the control flow a bit clearer (in my opinion):

```python
try:
    URL_VALIDATOR(redirect_url)
except ValidationError:
    return render_to_response('saml2idp/logged_out.html', {},
                              context_instance=RequestContext(request))
else:
    return HttpResponseRedirect(redirect_url)
```

(Also I noticed we're not using the return value of `URL_VALIDATOR` so we can drop `validator` in any case)